packages/hardhat-core/src/builtin-tasks/node.ts

```diff
@@ -66,6 +47,110 @@ Private Key: ${privateKey}
   }
 }
 
+subtask(TASK_NODE_GET_PROVIDER).setAction(
+  async (
+    {
+      forkBlockNumber: forkBlockNumberParam,
+      forkUrl: forkUrlParam,
+    }: {
+      forkBlockNumber?: number;
+      forkUrl?: string;
+    },
+    { config, network }
+  ): Promise<EthereumProvider> => {
+    // enable logging for in-memory hardhat network provider
+    await network.provider.request({
+      method: "hardhat_setLoggingEnabled",
+      params: [true],
+    });
+
+    assertHardhatInvariant(
```

Comment:
This shouldn't be present. We should create a provider IIF the network is not hardhat.

Not sure if we talked about this on slack.

This is useful to understand why: https://github.com/nomiclabs/buidler/pull/438#discussion_r376401348